src/msvc_helper_main-win32.cc

```diff
@@ -113,12 +113,13 @@ int MSVCHelperMain(int argc, char** argv) {
     PushPathIntoEnvironment(env);
   }
 
-  char* command = GetCommandLineA();
-  command = strstr(command, " -- ");
-  if (!command) {
+  wchar_t* w_command = GetCommandLine();
+  std::string w_command_str = WideToUtf8(w_command);
+  std::size_t command_pos = w_command_str.find(" -- ");
+  if(command_pos == std::string::npos){
     Fatal("expected command line to end with \" -- command args\"");
   }
-  command += 4;
+  std::string command = w_command_str.substr(command_pos + 4);
```

Comment:
This is a lot of string copying happening here.

Why not

char const * const command = w_command_str.c_str() + command_pos + 4;